s3transfer/futures.py

```diff
@@ -221,16 +222,23 @@ def result(self):
                 raise self._exception
             return self._result
 
-    def cancel(self, msg=''):
+    def cancel(self, msg='', by_user=True):
```

Comment:
Rather than `by_user`, what about providing your own cancelled error type.  That way we can extend this in the future for new cancellation types without having to add more params.